app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/remote/TasksRemoteDataSource.kt

```diff
@@ -80,22 +80,16 @@ object TasksRemoteDataSource : TasksDataSource {
         TASKS_SERVICE_DATA[task.id] = task
     }
 
-    override suspend fun completeTask(task: Task) {
-        val completedTask = Task(task.title, task.description, true, task.id)
-        TASKS_SERVICE_DATA[task.id] = completedTask
-    }
-
     override suspend fun completeTask(taskId: String) {
-        // Not required for the remote data source
-    }
-
-    override suspend fun activateTask(task: Task) {
-        val activeTask = Task(task.title, task.description, false, task.id)
-        TASKS_SERVICE_DATA[task.id] = activeTask
+        TASKS_SERVICE_DATA[taskId]?.apply {
+            isCompleted = true
```

Comment:
Having a model with mutable variables looks like a smell as changes to them won't cause new emissions to collectors.

We should change the `var`s in `Task` for `val`s unless there's a very important reason for not doing so

Reply:
That could be a TODO for a new PR, but this change will need to be reverted in there

Reply:
Agreed. This is a problem with the existing model, but exacerbated by my change. I will change in this PR.